### PR TITLE
feat: replicate boundary vectors to multiple partitions

### PR DESCRIPTION
**Pull request by BubbleCal**

### CODE REVIEW

File: rust/lance-index/src/vector/ivf/transform.rs [hunk 1]
```diff
@@ -106,6 +113,80 @@ impl<T: ArrowFloatType + L2 + Dot> IvfTransformer<T> {
 
         UInt32Array::from_iter(result.iter().flatten().copied())
     }
+
+    /// Compute the partition for each row in the input Matrix.
```
Comment (eddyxu) [inline rust/lance-index/src/vector/ivf/transform.rs:117]: do we use this in search? or just in build time.

**Review by chebbyChefNEQ — COMMENTED** [review]
just one concern: can we define `compute_partitions` in terms of `compute_multiple_partitions` instead of duplicating the code?

same for `compute_multiple_memberships`

File: rust/lance-index/src/vector/ivf/transform.rs (hunk 1 above)
Comment (BubbleCal) [reply rust/lance-index/src/vector/ivf/transform.rs:117]: only in build time

File: rust/lance-index/src/vector/ivf/transform.rs [hunk 2]
```diff
@@ -137,9 +218,31 @@ impl<T: ArrowFloatType + L2 + Dot + ArrowPrimitiveType> Transformer for IvfTrans
             })?;
 
         let mat = MatrixView::<T>::try_from(fsl)?;
-        let part_ids = self.compute_partitions(&mat).await;
-        let field = Field::new(PART_ID_COLUMN, part_ids.data_type().clone(), true);
-        Ok(batch.try_with_column(field, Arc::new(part_ids))?)
+        let part_ids = self.compute_multiple_partitions(&mat).await;
```
Comment (chebbyChefNEQ) [inline rust/lance-index/src/vector/ivf/transform.rs:221]: hmm, maybe we should use a different transformer for search?

**Comment by BubbleCal** [comment]
> just one concern: can we define `compute_partitions` in terms of `compute_multiple_partitions` instead of duplicating the code?
> 
> same for `compute_multiple_memberships`

sure we can, just duplicated the code for now because `compute_partitions` is extremely optimized and I paid as less as possible effort to get this feature worked first.
will remove the dup later

File: rust/lance-index/src/vector/ivf/transform.rs (hunk 2 above)
Comment (BubbleCal) [reply rust/lance-index/src/vector/ivf/transform.rs:221]: do we use transformer for search?
the code path should be `IVFIndex::find_partitions -> IvfImpl::find_partitions -> kmeans::find_partitions` right?

**Pull request closed without merging**